### DISCUSSION TITLE
fine-tune GPT2 for Question-Answering task

### DISCUSSION THREAD

**Discussion by AseelAlshorafa** [aub-mind/arabert#75] [Ideas]
First, I just want to thank you for your effort.
I saw the examples and you used Zero-Shot Question Answering I just want to ask if there is any way that I can fine-tune the model for question-answer datasets like ARCD,

**Comment by WissamAntoun** [comment]
Hey,
If you want the model to do question answering without looking at the context document, then you can finetune aragpt2 with the language modeling objective, i.e. you can train the model to predict the next token on data that has the same format as the one I used in the Zero-Shot examples but instead you add the real answer.

Or if you want to do a Squad like fine-tuning (which I don't think will work as good as AraBERT), then take a look at this repository https://github.com/ftarlaci/GPT2sQA

**Reply by AseelAlshorafa** [reply]
That was very helpful.
